ssh/ssh.go

```diff
@@ -100,41 +100,18 @@ func getVariant(osEnv config.Environment, gitEnv config.Environment, basessh str
 	return autodetectVariant(osEnv, gitEnv, basessh)
 }
 
-// findRuntimeDir returns a path to the runtime directory if one exists and is
-// guaranteed to be private.
-func findRuntimeDir(osEnv config.Environment) string {
-	if dir, ok := osEnv.Get("XDG_RUNTIME_DIR"); ok {
-		return dir
-	}
-	return ""
-}
-
 func getControlDir(osEnv config.Environment) (string, error) {
 	tmpdir, pattern := "", "sock-*"
 	if runtime.GOOS == "darwin" {
 		// On Darwin, the default temporary directory results in a socket path that's too long.
 		tmpdir = "/tmp"
 	}
-	dir := findRuntimeDir(osEnv)
-	if dir == "" {
-		return os.MkdirTemp(tmpdir, pattern)
-	}
-	dir = filepath.Join(dir, "git-lfs")
-	err := os.Mkdir(dir, 0700)
-	if err != nil {
-		// Ideally we would use errors.Is here to check against
-		// os.ErrExist, but that's not available on Go 1.11.
-		perr, ok := err.(*os.PathError)
-		if !ok || perr.Err != syscall.EEXIST {
-			return os.MkdirTemp(tmpdir, pattern)
-		}
-	}
-	return dir, nil
+	return os.MkdirTemp(tmpdir, pattern)
```

Comment:
There's no explanation here on why we've removed the functionality to store sockets in `XDG_RUNTIME_DIR`.  Unlike `/tmp`, that's guaranteed to live as long as the user's session and won't get cleaned up by a temporary file cleaner until the system reboots.  It's also guaranteed to be private and on a file system that is fully functional; that is, it supports sockets.

If we need to pick a random socket or create a temporary directory in that directory, I'd rather we do that than remove the `XDG_RUNTIME_DIR` functionality.

Reply:
> If we need to pick a random socket or create a temporary directory in that directory

I basically need a unique socket for every instance of `SSHTransfer`. Making a temporary directory in `XDG_RUNTIME_DIR` would probably do the trick.

> I'd rather we do that than remove the XDG_RUNTIME_DIR functionality.

Sure, I could do that.